ldk/javascript/src/search/index.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  promisifyMappedWithTwoParams,
+  promisifyMappedWithThreeParams,
+  promisifyWithParam,
+} from '../promisify';
+import * as mapper from './mapper';
+import { Document, Config, SearchResult } from './types';
+
+export * from './types';
+
+export interface Index {
```

Comment:
Should this be moved to `./types.ts`?

Reply:
My thought for this was `Index` and `Search` are interfaces with methods only and the other types were interfaces with properties only, but I can move it if you think it makes more sense there.

Reply:
I see your point. I don't think we ever clearly defined what goes where as a team. I usually just leave an interface which describes the top level of the aptitude and place rest in the types. But I don't see big issue here regardless.